ctapipe/reco/HillasReconstructor.py

```diff
@@ -212,46 +211,77 @@ def initialize_hillas_planes(
             dictionary of pointing direction per each telescope
         array_pointing: SkyCoord[AltAz]
             pointing direction of the array
+
+        Notes
+        -----
+        The part of the algorithm taking into account divergent pointing
+        mode and the correction to the psi angle is explained in [gasparetto]_
+        section 7.1.4.
         """
 
         self.hillas_planes = {}
         k = next(iter(telescopes_pointings))
         horizon_frame = telescopes_pointings[k].frame
         for tel_id, moments in hillas_dict.items():
-            # we just need any point on the main shower axis a bit away from the cog
-            p2_x = moments.x + 0.1 * u.m * np.cos(moments.psi)
-            p2_y = moments.y + 0.1 * u.m * np.sin(moments.psi)
-            focal_length = subarray.tel[tel_id].optics.equivalent_focal_length
 
             pointing = SkyCoord(
                 alt=telescopes_pointings[tel_id].alt,
                 az=telescopes_pointings[tel_id].az,
                 frame=horizon_frame,
             )
 
-            camera_frame = CameraFrame(
-                focal_length=focal_length, telescope_pointing=pointing
-            )
+            focal_length = subarray.tel[tel_id].optics.equivalent_focal_length
 
-            cog_coord = SkyCoord(x=moments.x, y=moments.y, frame=camera_frame,)
-            cog_coord = cog_coord.transform_to(horizon_frame)
+            if moments.x.unit == u.m:  # Image parameters are in CameraFrame
 
-            p2_coord = SkyCoord(x=p2_x, y=p2_y, frame=camera_frame)
-            p2_coord = p2_coord.transform_to(horizon_frame)
+                # we just need any point on the main shower axis a bit away from the cog
+                p2_x = moments.x + 0.1 * u.m * np.cos(moments.psi)
```

Comment:
This hard coded 0.1 m still bugs me, as it might not be "a bit away" for example for the SST camera. For the SST Camera, 0.1 * m is the camera radius.

Now that geometry has the cached `guess_camera_radius` property, maybe better would be to use `0.1 * cam.guess_camera_radius()`?

Reply:
Another advantage would be that the code would be the same for telescope mode and camera mode, as the camera mode would simply return the camera diameter in degrees and no unit is needed explicitly here.

Reply:
the difference with camera frame would still apply because x and y are flipped between the two

Reply:
just make sure you cache the result of `guess_camera_radius()`! Wouldn't want to compute that for every event.

Reply:
> the difference with camera frame would still apply because x and y are flipped between the two

We should fix that instead